decompiler/astdump.py

```diff
@@ -107,7 +107,7 @@ def print_list(self, ast):
 
     def print_dict(self, ast):
         # handles the printing of dictionaries
-        if type(ast) != dict:
+        if type(ast) is not dict:
```

Comment:
I'm not sure about these. The issue here is that `is` isn't overrideable, while == and != are. This is important, as the object on the right might be a `FakeClass` instance, and therefore of type `FakeClass`. Those class objects have overridden == and != methods, so they compare correct with instances of the actual type being presented.

Looking at it further these are all builtin types though that are special cased by the pickling machinery to begin with, so we should never be encountering those as a `FakeClass`. But just to warn you, with this project this transform isn't always safe.

Reply:
You know the code much better than I do, especially if there are some trickery happens or the magic/corrupy stuff involved is, where i largely draw a blank. 
I assumed for this cases the standard rule for comparisons. I did even think about if they should be `isinstance` comparisons instead `type`, but wasn't sure if this here intended is.

If there is risk this errors sometimes i will not take it just to satisfy some pep and the linter. So I'm going to revert them. Two can stay i think, but i ask for them better on their codeline.

Reply:
>I did even think about if they should be isinstance comparisons instead type, but wasn't sure if this here intended is.

Now that would be incorrect here ;). the lack of isinstance here is on purpose as the precondition for that function even being called already is `isinstance(ast, dict)`. An important thing to know about `isinstance` is that instances of a subclass of something also count as instances of that thing. And in this line of code, we do care about that difference, as for bare dicts we just want to print the dict, while for subclasses we want to print some extra information. So that's why we're directly checking the type of the variable.